src/screens/SiteDetailScreen.js

```diff
@@ -9,19 +9,85 @@ const {Marker} = MapView;
 
 import _ from 'lodash';
 
+import {attemptToAddFavorite, attemptToRemoveFavorite} from '../actions';
 
 import {linkColorBlue, navyBlueButton} from '../styles/index';
 
-import {navKeys, facilityIconDetails, featureIconDetails} from '../constants';
+import {navKeys, facilityIconDetails, featureIconDetails, map, tokens} from '../constants';
 import {site_detail_screen, campsite, common} from '../locale.en';
 import {campsiteIcon} from "../styles";
 
-const {header_title} = site_detail_screen;
 const {campsite_form} = campsite;
 const {location} = common;
 
 class SiteDetailScreen extends Component {
 
+    componentDidMount() {
+        const {selectedSite, currentUser, navigation: {setParams}} = this.props;
+
+        if (currentUser.name === tokens.GUEST) {
+            return;
+        }
+
+        const isFavorite = !!_.find(currentUser.favorites, favorite => favorite.id === selectedSite.id);
+
+        setParams({selectedSite, isFavorite, currentUser, toggleSiteFavorite: this.toggleSiteFavorite});
+    }
+
+    componentWillReceiveProps(nextProps) {
+        const {navigation: {setParams, state: {params}}} = this.props;
+
+        if (!_.isUndefined(params)) {
+            const {currentUser, selectedSite} = nextProps;
+            const willBeFavorite = !!_.find(currentUser.favorites, favorite => favorite.id === selectedSite.id);
+
+            if(willBeFavorite !== params.isFavorite){
+                setParams({isFavorite: willBeFavorite});
+            }
```

Comment:
Again, as always, need to be super careful about not falling into an infinite loop of `componentWillReceiveProps`, so we check to see if the next favorite state is different than our current one, and only then update the params (which update the favorite button on the top right nav)